### PR TITLE
Format requests/responses ahead of time in file audit

### LINKED ISSUE #7014: Performance Bottlecheck when Auditing to File
**Describe the bug**
When Vault is configured to audit to a file, the throughput of every Vault instance is severely impacted.

**To Reproduce**
1. Enable file auditing.  Run `vault audit enable file file_path=/dev/null`. 
2. Add a k/v secret. Run `vault kv put secret/foo bar=test`.
3. Read the secret at a high RPS.  Run `vault kv get secret/foo`.

**Expected behavior**
When reading a single k/v secret and auditing to the `/dev/null` device, I expect to see relatively high CPU utilization and high throughput of requests served. Under this configuration however, it is not possible to exceed around 30% CPU utilization on our hardware.

**Environment:**
* Vault Server Version (retrieve with `vault status`): Vault Enterprise 1.1.2
* Vault CLI Version (retrieve with `vault version`): N/A
* Server Operating System/Architecture: 
```
Distributor ID:	Debian
Description:	Debian GNU/Linux 8.11 (jessie)
Release:	8.11
Codename:	jessie
```

Vault server configuration file(s):

```hcl
storage "consul" {}

listener "tcp" {
  address       = "0.0.0.0:8200"
  tls_cert_file = "/etc/vault.d/cert.pem"
  tls_key_file  = <scrubbed>
}

seal "transit" {
  // The Vault token to use for unseal is injected by the entrypoint script.
  // The key_name and address are set as environment variables.
  disable_renewal    = "true"
  mount_path         = "transit/"
}

disable_sealwrap = "true"
plugin_directory = "/vault/plugins/"
```

### PR DESCRIPTION
**Pull request by jefferai**
This was inspired by #7022 but has the advantage of avoiding
double-locking and needing to perform lock upgrades while also
simplifying the logic and being faster.

Original, #7022, this:
```
goos: linux
goarch: amd64
pkg: github.com/hashicorp/vault/builtin/audit/file
BenchmarkAuditFile_request-4       30000             60734 ns/op
PASS
ok      github.com/hashicorp/vault/builtin/audit/file   2.428s

goos: linux
goarch: amd64
pkg: github.com/hashicorp/vault/builtin/audit/file
BenchmarkAuditFile_request-4       50000             34772 ns/op
PASS
ok      github.com/hashicorp/vault/builtin/audit/file   2.086s

goos: linux
goarch: amd64
pkg: github.com/hashicorp/vault/builtin/audit/file
BenchmarkAuditFile_request-4       50000             25302 ns/op
PASS
ok      github.com/hashicorp/vault/builtin/audit/file   1.542s
```
Fixes #7014
Closes #7022

### CODE REVIEW

File: builtin/audit/file/backend.go [hunk 1]
```diff
@@ -137,100 +144,118 @@ type Backend struct {
...
+	if _, err := reader.WriteTo(writer); err == nil {
+		b.fileLock.Unlock()
 		return nil
+	} else if b.path == "stdout" {
+		b.fileLock.Unlock()
+		return err
 	}
 
-	// Opportunistically try to re-open the FD, once per call
+	// If writing to stdout there's no real reason to think anything would have
```
Comment (vishalnayak) [inline builtin/audit/file/backend.go]: Do we want to log the error that we are ignoring here, as a warning in server logs?

Comment (vishalnayak) [reply builtin/audit/file/backend.go]: Nvm

Comment (jefferai) [reply builtin/audit/file/backend.go]: Sure, we could.

Comment (jefferai) [reply builtin/audit/file/backend.go]: Actually -- we can't. We don't have a logger at this point. We could maybe add that in centrally at some point in the future.

**Pull request merged**